cluster/state.go

```diff
@@ -292,3 +284,43 @@ func GetStateFromNodes(ctx context.Context, kubeCluster *Cluster) *Cluster {
 	log.Infof(ctx, "[state] Successfully fetched cluster state from Nodes")
 	return &currentCluster
 }
+
+func buildFreshState(ctx context.Context, kubeCluster *Cluster, newState *FullState) error {
+	rkeConfig := &kubeCluster.RancherKubernetesEngineConfig
+	// Get the certificate Bundle
+	certBundle, err := pki.GenerateRKECerts(ctx, *rkeConfig, "", "")
+	if err != nil {
+		return fmt.Errorf("Failed to generate certificate bundle: %v", err)
+	}
+	newState.DesiredState.CertificatesBundle = certBundle
+	if isEncryptionEnabled(rkeConfig) {
+		if newState.DesiredState.EncryptionConfig, err = kubeCluster.getEncryptionProviderFile(); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func rebuildExistingState(ctx context.Context, kubeCluster *Cluster, oldState, newState *FullState, flags ExternalFlags) error {
+	rkeConfig := &kubeCluster.RancherKubernetesEngineConfig
+	pkiCertBundle := oldState.DesiredState.CertificatesBundle
+	// check for legacy clusters prior to requestheaderca
+	if pkiCertBundle[pki.RequestHeaderCACertName].Certificate == nil {
+		if err := pki.GenerateRKERequestHeaderCACert(ctx, pkiCertBundle, flags.ClusterFilePath, flags.ConfigDir); err != nil {
+			return err
+		}
+	}
+	if err := pki.GenerateRKEServicesCerts(ctx, pkiCertBundle, *rkeConfig, flags.ClusterFilePath, flags.ConfigDir, false); err != nil {
+		return err
+	}
+	newState.DesiredState.CertificatesBundle = pkiCertBundle
+	if oldState.DesiredState.EncryptionConfig != "" && isEncryptionEnabled(rkeConfig) {
```

Comment:
Can we move this to a `isEncryptionEnabled(rkeConfig)` and then `if oldState.DesiredState.EncryptionConfig != "" {} else {}` ?